src/main/java/com/musala/atmosphere/server/websocket/ServerDispatcher.java

```diff
@@ -0,0 +1,353 @@
+     * 
+     * @param serverAddress
+     *        - an IP address for the WebSocket connection
+     * @param websocketPort
+     *        - a port for the WebSocket connection
+     */
+    public void startWebSocketServer(String serverAddress, int websocketPort) {
+        server = new org.glassfish.tyrus.server.Server(serverAddress,
+                                                       websocketPort,
+                                                       null,
+                                                       null,
+                                                       ClientServerEndpoint.class,
+                                                       ServerAgentEndpoint.class);
+        LOGGER.info("Websocket Server started on port " + websocketPort);
+
+        try {
+            server.start();
+        } catch (DeploymentException e) {
+            LOGGER.error("Could not start WebSocket server.", e);
+        }
+    }
+
+    /**
+     * Stops the WebSocket server.
+     */
+    public void stopWebsocketServer() {
+        if (server != null) {
+            server.stop();
+        }
+    }
+
+    /**
+     * Registers the connected agent on the server. Saves the agent's session for later use.
+     *
+     * @param registerAgentRequest
+     *        - {@linkplain RequestMessage} request message for a registration
+     * @param agentSession
+     *        - the session of the agent
+     */
+    void registerAgent(RequestMessage registerAgentRequest, Session agentSession) {
+        Object[] params = registerAgentRequest.getArguments();
+        String agentId = (String) params[0];
+        DeviceInformation[] devicesInformation = (DeviceInformation[]) params[1];
+
+        agentIdToAgentSessionCache.put(agentId, agentSession);
+        serverManager.registerAgent(agentId);
+
+        serverManager.publishAllDevicesForAgent(devicesInformation, agentId);
+        LOGGER.debug("Agent with ID " + agentId + " registered.");
+    }
+
+    /**
+     * Gets called when something in the device list has changed.
+     *
+     * @param deviceListChangedRequest
+     *        - the device list changed request message
+     */
+    void deviceListChanged(RequestMessage deviceListChangedRequest) {
+        DeviceInformation deviceInformation = (DeviceInformation) deviceListChangedRequest.getArguments()[0];
+        boolean isConnected = (boolean) deviceListChangedRequest.getArguments()[1];
+        String agentId = deviceListChangedRequest.getAgentId();
+        String deviceSerial = deviceListChangedRequest.getDeviceId();
+
+        try {
+            serverManager.onAgentDeviceListChanged(agentId, deviceSerial, deviceInformation, isConnected);
+        } catch (CommandFailedException ex) {
+            Session agentSession = agentIdToAgentSessionCache.get(agentId);
+            sendErrorResponseMessage(ex, agentSession, deviceListChangedRequest.getSessionId());
+            ex.printStackTrace();
```

Comment:
be consistent to your approach to exceptions - sometimes you print stack trace, sometimes you use LOGGER. I believe the latter is better